.ado/templates/e2e-testing-macos.yml

```diff
@@ -0,0 +1,34 @@
+steps:
+  - script: |
+      yarn appium driver install mac2
+    workingDirectory: apps/fluent-tester
+    displayName: 'Installing appium mac2 driver'
+
+  - script: |
+      yarn macos
+    workingDirectory: apps/fluent-tester
+    displayName: 'start the server'
```

Comment:
Technically this command both builds and runs the test app AND starts the server

Reply:
Ugh, I know. I need to add a comment explaining the reasoning behind this (same issue for Windows).

Basically, if I run `yarn start`, the command prompt runs continuously because there's no exit command. It won't process the next step.

This way, it opens up a second command prompt for the react-native server, rather than running it on the main one.

I'm sure there may be a better way to do this, but when I first implemented the Windows testing I couldn't find one. Maybe we can revisit it again

Reply:
At least in bash, you can add "&" to any job to background it. So "yarn start &" would run it in the background. I remember I experimented with that when we set up the Apple CI checks. I ultimately decided not to go with it because I didn't like the idea of starting background jobs and never closing them.

https://stackoverflow.com/questions/44222883/run-a-shell-script-and-immediately-background-it-however-keep-the-ability-to-in

Reply:
Good to know though, that this CI check relies on `yarn macos`  auto-launching the server. I had played around with a change where we don't do that. Glad I kept it in 😅

Reply:
Hmm, I didn't know about the "&" syntax. Maybe I'll try it. I wonder if we can force quit that terminal after we're done (or if it's even necessary, since E2E testing is the last thing that's run, and then the job is over)

Reply:
Could probably find the job ID and send a `kill` command. Or just let ADO destroying the VM handle it for us :) 

Or just not worry about it, and just run `yarn macos` :D

Reply:
Looks like I didn't even need to start the server - thanks for the tip. Resolving this now :)